client/data/newsletter-categories/use-newsletter-categories-query.tsx

```diff
@@ -0,0 +1,58 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type NewsletterCategoryQueryProps = {
+	blogId: number;
+};
+
+type NewsletterCategoryQueryResponse = {
+	newsletter_categories: NewsletterCategoryResponse[];
+};
+
+type NewsletterCategories = {
+	newsletterCategories: NewsletterCategory[];
+};
+
+type NewsletterCategoryResponse = {
+	term_id: number;
+	name: string;
+	description: string;
+	order: number;
+};
+
+type NewsletterCategory = {
+	termId: number;
+	name: string;
+	description: string;
+	order: number;
+};
+
+const convertNewsletterCategoryResponse = ( response: NewsletterCategoryQueryResponse ) => {
+	return {
+		newsletterCategories: response.newsletter_categories.map(
```

Comment:
You just return this map `response.newsletter_categories.map` instead instead of wrapping on another object, it would make it easier to consume without destructuring the result.

Reply:
What if we want to add more data to the response? Now it's just newsletter_categories, but tomorrow it could be newsletter_categories count, etc... If we are going to consume just the array of categories, why don't we make the endpoint to return the array of categories instead of a wrapper object like { newsletter_categories: ... } ?